### PR TITLE
Show ignore message in console and json output

### PR DESCRIPTION
**Pull request by yanganto**
- Provide ignore the message in console and JSON output
- Modify the ignore message style in the log file

related: #92714

### CODE REVIEW

**Comment by yanganto** [comment]
Hi @m-ou-se,
it may not be good and waste rust developing power to have more than one experience member to take a look at one thing. It is just a kindly notice to let you know it is fixing and I really thank you for the thing to check after the previous PR goes into nightly.  That is the thing I should do.  :pray: :heart:

**Comment by yanganto** [comment]
Hi @Mark-Simulacrum, do you have any concerns about this?  The ignore message will show as we think on the summary of `cargo test`.  If there is still something missing, please let me know. Thanks.

**Comment by Mark-Simulacrum** [comment]
I plan to review it soon, just haven't had the chance to yet.

**Comment by yanganto** [comment]
If there is something I can do, please let me know, I am willing to help. :)

File: library/test/src/formatters/pretty.rs [hunk 1]
```diff
@@ -59,10 +63,10 @@ impl<T: Write> PrettyFormatter<T> {
 
     pub fn write_short_result(
         &mut self,
-        result: &str,
+        result: impl AsRef<str>,
```
Comment (m-ou-se) [inline library/test/src/formatters/pretty.rs:66]: Why did this change? `&String` should coerce implicitly to `&str`.

File: library/test/src/formatters/pretty.rs [hunk 2]
```diff
@@ -45,8 +45,12 @@ impl<T: Write> PrettyFormatter<T> {
         self.write_short_result("FAILED", term::color::RED)
     }
 
-    pub fn write_ignored(&mut self) -> io::Result<()> {
-        self.write_short_result("ignored", term::color::YELLOW)
+    pub fn write_ignored(&mut self, may_message: Option<&'static str>) -> io::Result<()> {
+        if let Some(message) = may_message {
```
Comment (m-ou-se) [inline library/test/src/formatters/pretty.rs:49]: What does 'may' stand for? For 'maybe' I suppose? The `Option` type already makes it clear that the message is optional. There's no need to repeat that information in the name.
```suggestion
    pub fn write_ignored(&mut self, message: Option<&'static str>) -> io::Result<()> {
        if let Some(message) = message {
```

**Review by m-ou-se — COMMENTED** [review]
Thanks! Looks good. I left a few small comments.

**Pull request merged**